packages/ag-charts-website/src/utils/generateExample.ts

```diff
@@ -39,9 +40,57 @@ export async function generateExample({ exampleName, theme, format }: Params) {
         mockText: false,
     });
 
+    const options = {
+        animation: { enabled: false },
+        document: jsdom.window.document,
+        window: jsdom.window,
+        width: DEFAULT_THUMBNAIL_WIDTH,
+        height: DEFAULT_THUMBNAIL_HEIGHT,
+    } as any;
+
+    const chartProxy = AgCharts.create(options);
+
+    return [jsdom, mockCtx, chartProxy] as const;
+}
+
+const pool: ReturnType<typeof buildPoolEntry>[] = [];
+
+function initPool() {
+    // eslint-disable-next-line no-console
+    console.log(`Creating thumbnail pool of size ${THUMBNAIL_POOL_SIZE}`);
+    for (let i = 0; i < THUMBNAIL_POOL_SIZE; i++) {
+        pool.push(buildPoolEntry());
+    }
+}
+initPool();
+
+async function borrowFromPool() {
+    let count = 0;
+    while (pool.length === 0 && count < 5) {
+        // eslint-disable-next-line no-console
+        console.log('Waiting for pool to become available...');
+        await new Promise((resolve) => setTimeout(resolve, (10 * count) ^ 2));
+        count++;
+    }
+
+    if (pool.length === 0) {
+        throw new Error('No JSDOM instance available to borrow.');
+    }
+
+    return pool.splice(0, 1)[0];
```

Comment:
Is this not the same as:-

```suggestion
    return pool.shift();
```